portal/static/js/vendor/bootstrap_datatables_extension.js

```diff
@@ -38,33 +38,39 @@ var tnthTables = {
     "dateSorter": function(a, b) {
         a = a || 0;
         b = b || 0;
+        var a_d = 0, b_d = 0;
         /*
          * make sure the string passed in does not have line break element if so it is a possible mult-line text, split it up and use the first item in the resulting array
          */
         var regex = /<br\s*[\/]?>/gi;
-        a = a.replace(regex, "\n");
-        b = b.replace(regex, "\n");
-        var ar = a.split("\n");
-        if (ar.length > 0) {
-            a = ar[0];
+        //NOTE: adding if clause here because a is set to 0 if null/undefined, attempting to perform string specific operations,e.g. regular expression matching as follows, on 0, a numeric variable will result in runtime error
+        if (a) {
+            a = a.replace(regex, "\n");
```

Comment:
Should this logic be run when populating the table data, instead of sorting it? Wouldn't this change only affect the sort order if there's a leading `\n` in a cell?

Reply:
hmm, I believe, at least my understanding is that this part of code is called when the column is actively sorted via either user interaction on the UI or manual triggering , so this in fact _is_ dealing specifically with sorting  - so yes, it only affects sort order and would have been invoked **after** the table data has already been populated.